More changes after the calval release

gshiroma opened the pull request:
Changes:
- Rename static layers product from RTC-S1-SL to RTC-S1-STATIC;
- Remove rasters X, Y, and Inc from call to topo when creating the layover/shadow mask.
- Reduce radar grid dimensions when creating static layers.
- Multiple changes to the metadata.
- Bug fixes.

seongsujeong reviewed (approved):
Looks like the most of the changes in this PR are to deal with the features merged in ISCE3 side. The PR looks good to me.

seongsujeong commented on src/rtc/h5_prep.py:
```diff
@@ -667,39 +701,98 @@ def get_metadata_dict(product_id: str,
-        'metadata/processingInformation/parameters/postProcessingFilteringApplied':  # 1.7.4
-            ['post_processing_filtering_applied',
-             STANDARD_RTC_S1_ONLY,
+        'metadata/processingInformation/parameters/' +
```
Question: are `+` for readability or to concatenate the strings? If latter, we might not need the `+` sign here.

seongsujeong commented on src/rtc/h5_prep.py:
```diff
@@ -667,39 +701,98 @@ def get_metadata_dict(product_id: str,
+            ['processing_information'
+             '_output_backscatter_expression_convention',
+             ALL_PRODUCTS,
+             'linear power/intensity',
```
I think it has to be either `linear power` or `linear intensity`

gshiroma replied:
I think backscatter (coefficient) is a measurement of intensity, i.e., power per area, but one can also consider it the power received for a give radar bin. Usually, power and intensity are used interchangeably meaning that you take the square of the amplitude (volts). In CARD specifications they list two options: "linear amplitude" or "linear power", but since I prefer "intensity", I put "linear power/intensity". Maybe I can just go ahead and use "intensity". Any thoughts? @hfattahi , if you have any preference, please let us know.

seongsujeong replied:
Thanks for the explanation. I think the point of the CARD 3.2 (scaling conversion) is to clarify if the values are squared or not. To my understanding, both "intensity" and "power" are based on the square of "amplitude", so I thought  `power/intensity` are duplicating.

gshiroma replied:
Just updated it to "linear backscatter intensity"

seongsujeong commented on src/rtc/h5_prep.py:
```diff
@@ -667,39 +701,98 @@ def get_metadata_dict(product_id: str,
+             cfg_in.groups.processing.rtc.output_type,
+             'Backscatter normalization convention of this product (RTC-S1)'],
+        'metadata/processingInformation/parameters/' +
+        'outputBackscatterExpressionConvention':
```
```suggestion
        'outputBackscatterExpressionConvention':  # 3.1
```
to keep track of the CARD-NRB items

seongsujeong commented on src/rtc/h5_prep.py:
```diff
@@ -667,39 +701,98 @@ def get_metadata_dict(product_id: str,
+             'linear power/intensity',
+             'Backscatter expression convension'],
+        'metadata/processingInformation/parameters/' +
+        'outputBackscatterDecibelConversionEquation':
```
```suggestion
        'outputBackscatterDecibelConversionEquation':  # 3.2
```

seongsujeong commented on src/rtc/h5_prep.py:
```diff
@@ -667,39 +701,98 @@ def get_metadata_dict(product_id: str,
+             'Equation to convert provided backscatter to decibel (dB)'],
 
+        'metadata/processingInformation/parameters/geocoding/' +
+        'burstGeogridSnapX':
```
```suggestion
        'burstGeogridSnapX':  # 4.4
```

seongsujeong commented on src/rtc/h5_prep.py:
```diff
@@ -842,14 +941,29 @@ def get_metadata_dict(product_id: str,
-    if not is_mosaic:
+    if is_mosaic:
+        # Metadata only for the mosaic product
+        metadata_dict['metadata/processingInformation/parameters/geocoding/' +
```
Just a style suggestion. I think you would not need `+` here, but up to you.

seongsujeong commented on src/rtc/rtc_s1_single_job.py:
```diff
@@ -1821,15 +1823,16 @@ def run_single_job(cfg: RunConfig):
 
     if save_mosaics:
```
There is the same if statement right after this block, around line `1953` as below. Can you consider putting those two blocks into one?

```python
# Append metadata to mosaic GeoTIFFs
    if save_mosaics:
        for current_file in mosaic_output_file_list:
            if not current_file.endswith('.tif'):
                continue
            append_metadata_to_geotiff_file(current_file,
                                            mosaic_geotiff_metadata_dict,
                                            mosaic_product_id)

```

gshiroma replied:
@seongsujeong , sorry, I'm having a hard time understanding this comment. You mean that this block is duplicated? I couldn't find the duplicated code. Could you please elaborate?

seongsujeong replied:
Sorry for not being clear. what I meant is that I've found this pattern:

```python
if save_mosaics:  # line 1825
    [do something]

if save_mosaics:  #line 1954
    [do another thing]

```

So I just thought we can do like:

```python
if save_mosaics:
    [do something]
    [do another thing]

```

Please let me know if it makes sense.

gshiroma replied:
Ok, just updated it. Thanks

seongsujeong reviewed (commented):
Nice work. Here are my comments and suggestions - Seongsu

gshiroma commented on src/rtc/h5_prep.py:
```diff
@@ -382,6 +393,20 @@ def get_metadata_dict(product_id: str,
+        burst_snap_y = '(SNAP DISABLED FOR COORDINATE Y)'
+    mosaic_snap_x = cfg_in.groups.processing.mosaicking.mosaic_geogrid.x_snap
+    if not mosaic_snap_x:
+        mosaic_snap_x = '(SNAP DISABLED FOR COORDINATE X)'
```
Thank you, @seongsujeong . I think it's too wordy. I'm actually thinking about simplifying it to `DISABLED`.

seongsujeong reviewed (approved):
LGTM. Thanks for the addressing the comments.
Please rebase the PR #55 after merging this one, so that I can have a better look at the following PR.

gshiroma commented:
Thank you, @seongsujeong , for this quick review on a Saturday! I just addressed your comments. Please, let me know if I missed anything or if you have any other comments or concerns.
